### ISSUE TITLE
Another possible problem with rank file processing

### ISSUE THREAD

**Issue by gpaulsen** [open-mpi/ompi#6179]
@siegmargross is encountering problems using the rank file option while testing openmpi-master-201810190352-8db5aaa.  See https://www.mail-archive.com/users@lists.open-mpi.org/msg32797.html

He reported the following on Oct 22nd, 2018 with openmpi-master-201810190352-8db5aaa:
```
I've installed openmpi-master-201810190352-8db5aaa on my "SUSE Linux
Enterprise Server 12.3 (x86_64)" with Sun C 5.15 (Oracle Developer
Studio 12.6), gcc-6.4.0, icc-19.x, and pgcc-18.4 Unfortunately, I get
the following error for gcc, icc, and pgcc. I'm still unable to build
a version with Sun C due to the error that I reported some time ago
(https://users.open-mpi.narkive.com/lvQKElru/ompi-users-error-building-openmpi-master-201810050304-5f1c940-on-linux-with-sun-c).

loki config_files 139 head -7 /export2/src/openmpi-master/openmpi-master-201810190352-8db5aaa-Linux.x86_64.64_gcc/config.log | tail -1 $ ../openmpi-master-201810190352-8db5aaa/configure --prefix=/usr/local/openmpi-master_64_gcc --libdir=/usr/local/openmpi-master_64_gcc/lib64 --with-jdk-bindir=/usr/local/jdk-11/bin --with-jdk-headers=/usr/local/jdk-11/include JAVA_HOME=/usr/local/jdk-11 LDFLAGS=-m64 -L/usr/local/cuda/lib64 CC=gcc CXX=g++ FC=gfortran CFLAGS=-m64 -I/usr/local/cuda/include CXXFLAGS=-m64 -I/usr/local/cuda/include FCFLAGS=-m64 CPP=cpp -I/usr/local/cuda/include CXXCPP=cpp -I/usr/local/cuda/include --enable-mpi-cxx --enable-cxx-exceptions --enable-mpi-java --with-cuda=/usr/local/cuda --with-valgrind=/usr/local/valgrind --with-hwloc=internal --without-verbs --with-wrapper-cflags=-std=c11 -m64 --with-wrapper-cxxflags=-m64 --with-wrapper-fcflags=-m64 --enable-debug

loki config_files 139 ompi_info | grep "OPAL repo revision:"
      OPAL repo revision: v2.x-dev-6365-g8db5aaa

loki config_files 140 cat rf_loki
# Rankfile to bind processes to specific cores, sockets, ... This
# file can be used for openmpi-1.7.x or newer which use ";" to
# separate sockets and have different options than older versions.
#
#
# Usage:
#   mpiexec -report-bindings -np 1 -rf rf_loki hostname
#
#
# File: rf_loki                         Author: S. Gross
# Date: 04.08.2017
#

rank 0=loki slot=0:0;1:0

loki config_files 141 mpiexec -report-bindings -np 1 -rf rf_loki hostname
[loki:14821] OPAL dss:unpack: got type 27 when expecting type 3

[loki:14821] [[42585,0],0] ORTE_ERROR_LOG: Pack data mismatch in file ../../../../openmpi-master-201810190352-8db5aaa/orte/mca/odls/base/odls_base_default_fns.c at line 625

loki config_files 142
```

**Comment by rhc54** [comment]
Probably addressed by https://github.com/open-mpi/ompi/pull/6203

**Comment by jsquyres** [comment]
@siegmargross This should be fixed in master; awaiting merge to v4.0.x.

**Comment by gpaulsen** [comment]
I'm going to verify this on v4.0.x next week.
